constraints_test.go

```diff
@@ -209,6 +209,54 @@ func TestConstraintCheck(t *testing.T) {
 	}
 }
 
+func TestReleaseVsPrerelease(t *testing.T) {
+	tests := []struct {
+		constraint string
+		version    string
+		check      bool
+	}{
+		{"<1.0.0", "1.0.0-alpha", true},
+		{"<1.0.0", "1.0.0-beta1", true},
+		{"<1.0.0", "1.0.0-rc.1", true},
+		{"<1.0.0", "1.0.0-rc1", true},
+		{"<2.0.0-0", "1.0.0-alpha", true},
+		{"<2.0.0-0", "1.0.0-beta1", true},
+		{"<2.0.0-0", "1.0.0-rc.1", true},
+		{"<2.0.0-0", "1.0.0-rc1", true},
+		{"<2.0.0-rc.1", "1.0.0-alpha", true},
+		{"<2.0.0-rc.1", "1.0.0-beta1", true},
+		{"<2.0.0-rc.1", "1.0.0-rc.1", true},
+		{"<2.0.0-rc.1", "1.0.0-rc1", true},
+		{"<2.0.0", "1.0.0-alpha", true},
+		{"<2.0.0", "1.0.0-beta1", true},
+		{"<2.0.0", "1.0.0-rc.1", true},
+		{"<2.0.0", "1.0.0-rc", true},
+		{"<2.0.0", "1.0.0-rc1", true},
+		{"<2.0.0", "1.0.0", true},
+		{"<2.0", "1.0", true},
+		{"<2", "1", true},
```

Comment:
There are 2 problems with these tests...

1. They don't represent the intended behavior of v3. A change to produce this output cannot happen without a new major version.
2. The current behavior is [intentional](https://github.com/Masterminds/semver/issues/150#issuecomment-2310793333). Prereleases need to be handled differently from stable releases. Including them by default would produce some bad results. This was discovered with earlier versions of this library.

Tests being added need to follow the current intent.

Reply:
I didn't realize that fixing this was out of scope. Intentional or not, I've pointed out in a separate issue how the behavior doesn't match the spec.

It's fine. I can fork and fix. Thanks.

Reply:
@skyzyx There is no spec for semver constraint systems. If you look at the semver resource itself, it handle pre-releases out of the box. But, constraint systems like you see in each language and tools built on that deal with the practical realities of using semvers have no spec. If you disagree with that decision here, given that prereleases are unstable and not expected to meet the guarantees of stable releases, please contact projects like node-semver used by npm which does the same thing.